module5/TaskListApp/TaskListApp/Models/TasksStore.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  TasksStore.swift
+//  TaskListApp
+//
+//  Created by Peter Rice on 12/22/23.
+//
+
+import Foundation
+
+class TasksStore: ObservableObject {
+  let defaultTasks = [
+    Task(id: UUID(), title: "Task 1", isCompleted: false, notes: ""),
+    Task(id: UUID(), title: "Task 2", isCompleted: false, notes: ""),
+    Task(id: UUID(), title: "Task 3", isCompleted: false, notes: ""),
+    Task(id: UUID(), title: "Task 4", isCompleted: false, notes: ""),
+    Task(id: UUID(), title: "Task 5", isCompleted: true, notes: "some notes"),
+    Task(id: UUID(), title: "Task 6", isCompleted: false, notes: ""),
+    Task(id: UUID(), title: "Task 7", isCompleted: false, notes: ""),
+    Task(id: UUID(), title: "Task 8", isCompleted: false, notes: ""),
+    Task(id: UUID(), title: "Task 9", isCompleted: false, notes: ""),
+    Task(id: UUID(), title: "Task 10", isCompleted: false, notes: "")
+  ]
+  
+  @Published var tasks: [Task]
+  
+  init() {
+    tasks = defaultTasks
+  }
+  
+  func addTask(title: String, notes: String) {
+    tasks.append(Task(id: UUID(), title: title, isCompleted: false, notes: notes))
+  }
+    
+  func setTask(with id: UUID, completed: Bool) {
+    if var taskToUpdate = tasks.first(where: { $0.id == id}) {
+      taskToUpdate.isCompleted = completed
+      tasks = tasks.map({
+        $0.id == id ? taskToUpdate : $0
+      })
+    }
+  }
+}
+
+struct Task: Identifiable, Hashable {
```

Comment:
Because `id` and `isCompleted` are known values you could modify `Task`
````
struct Task: Identifiable, Hashable {
 let id = UUID()
var title: String
var isCompleted = false
var notes: String
}
````
That makes the initializers easier and shorter.

Reply:
Good suggestion. Actually going to set the default for isCompleted via an initializer default arg, so I can still easily configure it for mock and preview data.

Reply:
That makes sense for testability. You can also make an optional init method so you can do both, one with the default value and one with a passed in value. It might look like
````
struct Task: Identifiable, Hashable {
  let id = UUID()
  var title: String
  var isCompleted: Bool
  var notes: String

init(title: String, notes: String) {
self.title = title
self.notes = notes
isCompleted = false
}

init(title: String, notes:String: isCompleted: Bool) {
self.title = title
self.notes = notes
isCompleted = isCompeted
}